### ISSUE TITLE
Extend sub-tasks to track and log any retries attempts 

### ISSUE THREAD

**Issue by sambles** [OasisLMF/OasisPlatform#1097]
## Issue Description
The problem comes when a task fails, and is then successful on a follow up attempt (up to three). 
When this happens its not clear (from a user's view point) since northing is logged. 

### Approach 1 
Add a new section under sub-tasks to log any failed execution attempts. 

### Approach 2
Insert a duplicate sub-task marked as ERROR along with the failed logs. 

> **carlfischerjba:** The advantage of using the existing structure is that you could plot the failed tasks with the same command as all the others, and choose to change just the colour for instance. You'd end up with duplicates with the same task id appearing several times which might be a problem for some parsers.

> any tools reading those files would need to manage those cases correctly. There's also the question of whether the start time should be the start time of the first attempt or the final successful attempt. I would go for the latter since we're often interested in the time that a task was actively being run separately from the time that it was queued and waiting for hardware availability, but having start and end times for the failed attempts would complete the picture.
Maybe each task element could contain an additional list where each element of that list would contain the timestamps and the status of each retry. The task itself would still contain its final status.

**Comment by sambles** [comment]
**Side note on this issue:** If a sub-task takes **5h's**  and fails with an OOM error at **4:30h's**, then  we'd probably prefer to fail the analysis rather than waiting for **9h's** for two more retry failures. 

Suggestion is to add an execution time threshold.  e.g. if a task has been running for more than an hour (or whatever the threshold is) then fail a task on the first failure.

**Comment by carlfischerjba** [comment]
> **Side note on this issue:** If a sub-task generally takes **5h's** and fails with an OOM error at **4:30h's**, then we'd probably prefer to fail the analysis rather than waiting for **9h's** for two more retry failures.

Good point @sambles. The counterpoint is that the retries might lead to the memory-heavy task running alongside lighter-weight tasks or with fewer tasks running on the same machine and therefore succeeding (but there's no guarantee).

More generally, how often do models hit situations where a retry is helpful? I could see them being helpful for network glitches but most errors in the model itself would presumably just repeat themselves.
